File: mobile/src/release/java/jp/satorufujiwara/kotlin/AppModule.java
package jp.satorufujiwara.kotlin;

import com.squareup.leakcanary.RefWatcher;

import android.app.Application;
import android.support.annotation.NonNull;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final DaggerApp app;

    public AppModule(@NonNull final DaggerApp app) {
        this.app = app;
    }

    @Provides
    @AppScope
    Application provideApplication() {
        return app;
    }

    @Provides
    public AppLifecycleCallbacks provideAppLifecycleCallbacks() {
        return new ReleaseAppLifecycleCallbacks();
    }

    @Provides
    public RefWatcher provideRefWatcher() {
        return RefWatcher.DISABLED;
    }

}

File: mobile/src/release/kotlin/jp/satorufujiwara/kotlin/AppModule.kt
package jp.satorufujiwara.kotlin

import android.app.Application
import com.squareup.leakcanary.RefWatcher
import dagger.Module
import dagger.Provides

@Module
public class AppModule(private val app: Application) {

    @Provides
    @AppScope
    fun provideApplication(): Application = app

    @Provides
    @AppScope
    public fun provideAppLifecycleCallbacks(): AppLifecycleCallbacks = ReleaseAppLifecycleCallbacks()

    @Provides
    @AppScope
    public fun provideRefWatcher(): RefWatcher = RefWatcher.DISABLED

}
